### PR TITLE
simplify torch imports - Tensor & Module

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Notably, following 2 changes in the imports are made:

- `from torch import Tensor`
- `from torch import nn` and using `nn.ModuleDict, nn.Module, nn.ModuleList`

instead of: `import torch` and using:

- `torch.Tensor`
- `torch.nn.Module, torch.nn.ModuleDict, torch.nn.ModuleList`

## Before submitting
- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/utils.py
Comment (krshrimali) [inline flash/core/data/utils.py:23]: Thanks for the refactoring, @Borda - but I'm not very sure if this is a great idea. And this might be just me, I usually prefer having a namespace around such modules where it's not immediately clear where they come from. So something like `nn.ModuleDict` is more clear than `ModuleDict` and doing `from torch.nn import ModuleDict`.

**Review by krshrimali — APPROVED** [review]
Thanks, @Borda for the refactoring! I'm a little skeptical about the changes, in some places, but overall it looks fine. I'll wait for Ethan's comments, and then we can merge this if all is good :)

File: flash/core/data/utils.py
Comment (Borda) [reply flash/core/data/utils.py:23]: the long imports tend to cut lines also I believe import is slightly faster...

**Comment by krshrimali** [comment]
Hi, @Borda - I'm okay with replacing `torch.Tensor` with `Tensor`, but hesitant on replacing other imports. And this can be just me, but I've noticed that using `nn.Module` makes it clear that it comes from the `nn` module of PyTorch, while using `Module` doesn't tell a lot where it's coming from. Same goes for `ModuleDict` and other imports. Though, I'm fine if in a separate PR later we decide to refactor / simplify the imports for `torch.Tensor`, but I still think it's fine the way the code is right now. With these simplified imports (though we can not help much about torchvision as using `T.ToTensor()` like structure is quite standard), we get these errors like `T not defined` (https://github.com/Lightning-AI/lightning-flash/issues/1403) if the user doesn't have torchvision installed. Though, this has to do more with raising better error messages if the module is not available, but just sharing the issue here.

If you think otherwise, I'm happy to discuss this further. Please feel free to re-open it in that case.

Thanks!

**Comment by Borda** [comment]
> noticed that using `nn.Module` makes it clear that it comes from the `nn` module of PyTorch

well, do you know any other `Module` coming from a different library?

> I'm okay with replacing `torch.Tensor` with `Tensor`

so why close full PR and not ever the second change?

**Comment by krshrimali** [comment]
> > noticed that using `nn.Module` makes it clear that it comes from the `nn` module of PyTorch
> 
> well, do you know any other `Module` coming from a different library?

As I said, it makes it clear that it's coming from `nn`. It's not about whether there are ambiguities, it's `from library import *` vs `import library` and then using `library.*` practice. While both of them are right, it's more about what is preferred, and in my message, I just said that I would prefer doing `import library` and then doing `library.*`.

> > I'm okay with replacing `torch.Tensor` with `Tensor`
> 
> so why close full PR and not ever the second change?

Cool, I'll remove the other import changes, and keep `torch.Tensor` -> `Tensor` changes, but still I would say for consistency, either we should do: `from library import *` or `import library` -> `library.*` behavior.

Open to discussions on this. Thanks!

**Pull request merged**
